api/types/device.go

```diff
@@ -90,6 +90,15 @@ func UnmarshalDevice(raw []byte) (*DeviceV1, error) {
 	return dev, trace.Wrap(dev.CheckAndSetDefaults())
 }
 
+// MarshalDevice marshals a DeviceV1 resource.
+func MarshalDevice(dev *DeviceV1) ([]byte, error) {
+	devBytes, err := json.Marshal(dev)
```

Comment:
nit: we typically prefer using `utils.FastMarshal` over `json.Marshal`.  Using that would require moving this function to somewhere in `lib` since you can't import `lib/utils` from the `api/*` family of packages, but if there isn't a particular need for this function to live here, that might be worth doing.  Most typed marshal/unmarshal functions are defined in `lib/services` anyhow.

Reply:
I decided to keep it as is because the way device services are currently setup, there's nothing to be added to `lib/services` except for these marshal/unmarshal func. For other services in `lib/services`, we have interface defined for them and then the marshalers works with the interface. We already have two types for Devices (`Device` and `DeviceV1`) and adding device marshalers to `lib/services` in similar way will (IMO) only complicate things (more scattered code for Device trust).

Reply:
Moved device marshalers to `lib/services` https://github.com/gravitational/teleport/pull/24963/commits/6c4b9ad9b66c1b9411c4f7e917d25b6c0b623361